build.assets/tooling/cmd/notarize-apple-binaries/hashicorpLoggerAdapter.go

```diff
@@ -0,0 +1,228 @@
+}
+
+// Args are alternating key, val pairs
+// keys must be strings
+// vals can be any type, but display is implementation specific
+// Emit a message and key/value pairs at a provided log level
+func (hclla HCLogLogrusAdapter) Log(level hclog.Level, msg string, args ...interface{}) {
+	logrusLevel := hclogLevelToLogrusLevel(level)
+	msgWithImpliedArgs := fmt.Sprintf("%s %%s", msg)
+	argsWithImpliedArgsString := append(args, hclla.createImpliedArgsString())
+	hclla.logger.Logf(logrusLevel, msgWithImpliedArgs, argsWithImpliedArgsString)
+}
+
+// Emit a message and key/value pairs at the TRACE level
+func (hclla HCLogLogrusAdapter) Trace(msg string, args ...interface{}) {
+	hclla.Log(hclog.Trace, msg, args)
+}
+
+// Emit a message and key/value pairs at the DEBUG level
+func (hclla HCLogLogrusAdapter) Debug(msg string, args ...interface{}) {
+	hclla.Log(hclog.Debug, msg, args)
+}
+
+// Emit a message and key/value pairs at the INFO level
+func (hclla HCLogLogrusAdapter) Info(msg string, args ...interface{}) {
+	hclla.Log(hclog.Info, msg, args)
+}
+
+// Emit a message and key/value pairs at the WARN level
+func (hclla HCLogLogrusAdapter) Warn(msg string, args ...interface{}) {
+	hclla.Log(hclog.Warn, msg, args)
+}
+
+// Emit a message and key/value pairs at the ERROR level
+func (hclla HCLogLogrusAdapter) Error(msg string, args ...interface{}) {
+	hclla.Log(hclog.Error, msg, args)
+}
+
+// Indicate if TRACE logs would be emitted. This and the other Is* guards
+// are used to elide expensive logging code based on the current level.
+func (hclla HCLogLogrusAdapter) IsTrace() bool {
+	return hclla.logger.IsLevelEnabled(logrus.TraceLevel)
+}
+
+// Indicate if DEBUG logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsDebug() bool {
+	return hclla.logger.IsLevelEnabled(logrus.DebugLevel)
+}
+
+// Indicate if INFO logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsInfo() bool {
+	return hclla.logger.IsLevelEnabled(logrus.InfoLevel)
+}
+
+// Indicate if WARN logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsWarn() bool {
+	return hclla.logger.IsLevelEnabled(logrus.WarnLevel)
+}
+
+// Indicate if ERROR logs would be emitted. This and the other Is* guards
+func (hclla HCLogLogrusAdapter) IsError() bool {
+	return hclla.logger.IsLevelEnabled(logrus.ErrorLevel)
+}
+
+// ImpliedArgs returns With key/value pairs
+func (hclla HCLogLogrusAdapter) ImpliedArgs() []interface{} {
+	return hclla.impliedArgs
+}
+
+// Creates a sublogger that will always have the given key/value pairs
+func (HCLogLogrusAdapter) With(args ...interface{}) hclog.Logger {
+	// Ensure that there is a key for every value
+	if len(args)%2 != 0 {
+		extraValue := args[len(args)-1]
+		// Pulled from https://github.com/hashicorp/go-hclog/blob/main/intlogger.go#L731
+		args[len(args)-1] = "EXTRA_VALUE_AT_END"
+		args = append(args, extraValue)
+	}
+
+	newLogger := NewHCLogLogrusAdapter()
```

Comment:
Do you need to explicitly pass down the parent `logrus` logger here? If someone has customized that at some point this will reset the child HCLog to use the the default `logrus` logger.

Reply:
That's probably a good idea, especially if this gets put in a "library" module for our tooling. Updated